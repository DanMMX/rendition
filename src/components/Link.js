const get = require('lodash/get')
const omit = require('lodash/omit')
import * as React from 'react'
import styled from 'styled-components'
import { color, fontSize, space } from 'styled-system'
import { align, bold, caps } from './Txt'
import asRendition from '../asRendition'
import { darken, monospace } from '../utils'

let Base = styled.a`
  ${align}
  ${color}
  ${fontSize}
  ${monospace};
  ${space}

  ${caps}
  ${bold}

  text-decoration: ${props => props.decor || 'none'};
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  opacity: ${props => (props.disabled ? 0.65 : 1)};
  display: inline-block;

  &:active,
  &:hover {
    color: ${props =>
    !props.disabled &&
      darken(get(props.theme.colors, props.color) || props.color)};
  }
`

const Link = ({ is, blank, ...props }) => {
  if (is) {
    Base = Base.withComponent(is)
  }
  if (props.disabled) {
    props = omit(props, 'href')
  }
  return (
    <Base {...props} rel={blank && 'noopener'} target={blank && '_blank'} />
  )
}

Link.defaultProps = {
  is: 'a',
  color: 'primary.main'
}

export default asRendition(Link)
