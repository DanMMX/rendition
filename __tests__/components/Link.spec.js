import React from 'react';
import renderer from 'react-test-renderer';
import Provider from '../../src/components/Provider'
import Link from '../../src/components/Link'

test('Link renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <Link>Link</Link>
    </Provider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
