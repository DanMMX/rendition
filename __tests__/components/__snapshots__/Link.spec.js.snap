// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<div
  className="sc-bdVaJa jKopLr"
>
  <a
    className="sc-htoDjs gDCMNh sc-dnqmqq hHeMlD"
    color="primary.main"
    rel={undefined}
    target={undefined}
  >
    Link
  </a>
</div>
`;
